sql/src/main/java/org/apache/druid/sql/calcite/schema/InformationSchema.java

```diff
@@ -483,18 +479,21 @@ private static TableMacro getView(final SchemaPlus schemaPlus, final String func
       final AuthenticationResult authenticationResult
   )
   {
-    if (druidSchemaName.equals(subSchema.getName())) {
-      // The "druid" schema's tables represent Druid datasources which require authorization
+    final NamedSchema schema = namedSchemaMap.get(subSchema.getName());
+    if (schema != null && schema.getSchemaResourceType() != null) {
```

Comment:
nit: this functions code and getAuthorizedFunctionNamesFromSubSchema look very similar. Could do a slight refactor and reduce code here it seems. Not necessary though.